source/core.py

```diff
@@ -130,14 +129,43 @@ def handleReplaceCLIArg(cliArgument: str) -> bool:
 		if updateCheck and not config.conf['update']['askedAllowUsageStats']:
 			# a callback to save config after the usage stats question dialog has been answered.
 			def onResult(ID):
-				import wx
 				if ID in (wx.ID_YES,wx.ID_NO):
 					try:
 						config.conf.save()
 					except:
 						pass
 			# Ask the user if usage stats can be collected.
 			gui.runScriptModalDialog(gui.startupDialogs.AskAllowUsageStatsDialog(None), onResult)
+	addonFailureMessages: list[str] = []
+	failedUpdates = addonHandler._failedPendingInstalls.intersection(addonHandler._failedPendingRemovals)
+	failedInstalls = addonHandler._failedPendingInstalls - failedUpdates
+	failedRemovals = addonHandler._failedPendingRemovals - failedUpdates
+	if failedUpdates:
+		addonFailureMessages.append(
+			# Translators: Shown when one or more add-ons failed to update.
+			_("Following add-ons failed to update: {}").format(", ".join(failedUpdates))

```

Comment:
Should these messages use ngettext to support singular vs plural?

Reply:
Yes for using `ngettext`.

Also should we write "Following add-ons" or "The following add-ons"?